packages/combobox/__tests__/combobox.test.tsx

```diff
@@ -219,9 +220,36 @@ describe("<Combobox />", () => {
       expect(getByTextWithMarkup(optionToSelect)).toBeInTheDocument();
     });
   });
+
+  describe("Combobox inside dialog", () => {
+    it("should not close the dialog when Esc key is pressed", async () => {
+      let { getByTestId, queryByTestId } = render(<BasicDialog />);
+      let input = getByTestId("input");
+
+      await userEvent.type(input, "e");
+
+      expect(getByTestId("list")).toBeInTheDocument();
+
+      await userEvent.type(input, "{esc}");
+
+      expect(queryByTestId("list")).not.toBeInTheDocument();
```

Comment:
@frontsideair thanks for your PR contribution!

The issue with your test seems to be the use of `queryByTestId`. Using `ByTestId` is [not recommended][1] if there are other queries more inline with how the software is used by people or screen readers. We can use `ByRole` for most things associated with Combobox, and indeed your tests pass with these query selectors.

Given #395 also cares about the event propagating, we should also assert that our dialog remains open when escape is pressed. Perhaps something like this:

```js
it("should not close the dialog when Esc key is pressed", async () => {
  let { getByRole, queryByRole } = render(<BasicDialog />);
  let input = getByRole("combobox");

  expect(getByRole("dialog")).toBeInTheDocument();

  await userEvent.type(input, "e");

  expect(getByRole("listbox")).toBeInTheDocument();

  await userEvent.type(input, "{esc}");

  // escape should close our listbox
  expect(queryByRole("listbox")).not.toBeInTheDocument();

  // but our dialog should still be open
  expect(queryByRole("dialog")).toBeInTheDocument();
});
```

[1]: https://testing-library.com/docs/dom-testing-library/api-queries#bytestid